include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,191 @@
+                    return that.from_ == from_ && that.to_ == to_ && that.spacing_ == spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void next()
+                {
+                    if (spacing_ == T{0})
+                    {
+                      RANGES_EXPECT(to_ == from_);
+                      to_ += 1;
+                    }
+                    from_ += spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void prev()
+                {
+                    from_ -= spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void advance(std::ptrdiff_t n)
+                {
+                    
+                  RANGES_EXPECT(spacing_ == T{0} || std::numeric_limits<T>::max() / spacing_ >= n);
+                    RANGES_EXPECT(to_ - from_ >= n * spacing_);
+                    from_ += n * spacing_;
+                }
+                constexpr std::ptrdiff_t distance_to(linear_distribute_integral_view const &that) const
+                {
+                    return (that.from_ - from_) / spacing_;
+                }
+            public:
+                linear_distribute_integral_view() = default;
+                linear_distribute_integral_view(T from, T to, std::ptrdiff_t n)
+                  : from_(from), to_(to) 
+                {
+                    RANGES_EXPECT(n > 0);
+                    RANGES_EXPECT(to_ >= from_);
+                    if (to_ == from_)
+                    {
+                        RANGES_EXPECT(n == 1);
+                        spacing_ = 0;   
+                    }
+                    else
+                    {
+                        RANGES_EXPECT(n > 1);
+                        spacing_ = (to_ - from_)/ static_cast<T>(n - 1);   
+                    }
+                }
+                constexpr std::size_t size() const noexcept
+                {
+                    return spacing_ == T{0}? 1 : static_cast<std::size_t>((to_ - from_) / spacing_ + 1);
+                }
+            };
+
+            template <typename T>
+            struct linear_distribute_floating_point_view
+              : view_facade<linear_distribute_floating_point_view<T>, finite>
+            {
+                static_assert(std::is_floating_point<T>(), "");
+            private:
+                friend range_access;
+
+                T from_, to_;
+                std::ptrdiff_t n_;
+                
+                constexpr T read() const
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const
+                {
+                    return n_ == 0;
+                }
+                RANGES_CXX14_CONSTEXPR void next()
+                {
+                    RANGES_EXPECT(n_ > 0);
+                    from_ += (to_ - from_) / T(--n_);
+                }
+                RANGES_CXX14_CONSTEXPR void prev()
+                {
+                    from_ -= (to_ - from_) / T(++n_);
```

Comment:
I suspect there are corner cases where this does not perfectly invert the behavior of `next`. No correction yet suggested, but we need to be careful here.